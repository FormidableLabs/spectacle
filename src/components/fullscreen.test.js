import React from 'react';
import { mount } from 'enzyme';
import { mountToJson } from 'enzyme-to-json';
import Fullscreen from './fullscreen';

describe('<Fullscreen />', () => {
  test('should render correctly.', () => {
    const context = { styles: { styles: { fullscreen: {} } } };
    const wrapper = mount(<Fullscreen />, { context });
    expect(mountToJson(wrapper)).toMatchSnapshot();
  });

  test('should toggle fullscreen when the button is selected.', () => {
    const context = { styles: { styles: { fullscreen: {} } } };
    const wrapper = mount(<Fullscreen />, { context });
    wrapper.instance().handleToggleFullScreen = jest.fn();
    wrapper.update();
    wrapper.find('svg').simulate('click');
    expect(wrapper.instance().handleToggleFullScreen).toHaveBeenCalledTimes(1);
  });
});
