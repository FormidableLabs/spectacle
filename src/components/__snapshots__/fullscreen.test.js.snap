// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fullscreen /> should render correctly. 1`] = `
<Fullscreen>
  <svg
    data-radium={true}
    height="30px"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "bottom": 20,
        "cursor": "pointer",
        "opacity": 0,
        "position": "absolute",
        "right": 20,
        "transition": "300ms opacity ease",
      }
    }
    viewBox="0 0 512 512"
    width="30px"
  >
    <path
      d="M73.143,329.143H0V512h182.857v-73.143H73.143V329.143z M0,182.857h73.143V73.143h109.715V0H0V182.857z M438.857,438.857 H329.143V512H512V329.143h-73.143V438.857z M329.143,0v73.143h109.715v109.715H512V0H329.143z"
    />
  </svg>
</Fullscreen>
`;
